lib/blockchain.py

```diff
@@ -315,16 +325,75 @@ def segwit_support(self, N=144):
         h = self.local_height
         return sum([self.BIP9(h-i, 2) for i in range(N)])*10000/N/100.
 
+    def get_suitable_block_height(self, suitableheight):
+
+	#In order to avoid a block in a very skewed timestamp to have too much
+	#influence, we select the median of the 3 top most block as a start point
+	#Reference: github.com/Bitcoin-ABC/bitcoin-abc/master/src/pow.cpp#L201
+	blocks2height=suitableheight
+	blocks1height=suitableheight-1
+	blocksheight=suitableheight-2
+	blocks2 = self.read_header(blocks2height)
+	blocks1 = self.read_header(blocks1height)
+	blocks = self.read_header(blocksheight)
+	if (blocks['timestamp'] > blocks2['timestamp'] ):
+		blocksheight,blocks2height = blocks2height,blocksheight
+	if (blocks['timestamp'] > blocks1['timestamp'] ):
+		blocksheight,blocks1height = blocks1height,blocksheight
+	if (blocks1['timestamp'] > blocks2['timestamp'] ):
+		blocks1height,blocks2height = blocks2height,blocks1height
+	return blocks1height
+
+
     def get_median_time_past(self, height):
         times = [self.read_header(h)['timestamp']
                  for h in range(max(0, height - 10), height + 1)]
         return sorted(times)[len(times) // 2]
+ 
 
     def get_bits(self, height):
         '''Return bits for the given height.'''
         if bitcoin.TESTNET:
-            return 0
-        # Difficulty adjustment interval?
+            return 0 
+	
+	#NOV 13 HF DAA
+        daa_mtp=self.get_median_time_past(height-1)
+	
+	#######FOR TESTING        
+	#print ("debug, mtp is ",daa_mtp)
+	#print ("VALIDATING BLOCK HEIGHT ",height)
+        #######if (daa_mtp > 1509333800):  #leave this here for testing
+        if (daa_mtp > 1510600000):
+            daa_starting_height=self.get_suitable_block_height(height-144)
+            daa_ending_height=self.get_suitable_block_height(height)
+            daa_cumulative_work=0
+            daa_starting_timestamp=0
+            daa_ending_timestamp=0
+            for daa_i in range (daa_starting_height,daa_ending_height+1):
+                daa_prior = self.read_header(daa_i)
+                if (daa_i == daa_ending_height):
+                    daa_ending_timestamp=daa_prior['timestamp']
+                if (daa_i == daa_starting_height):
+                    daa_starting_timestamp=daa_prior['timestamp']
+                daa_bits_for_a_block=daa_prior['bits']
+                daa_work_for_a_block=bits_to_work(daa_bits_for_a_block)
+                daa_cumulative_work += daa_work_for_a_block
+	    daa_elapsed_time=daa_ending_timestamp-daa_starting_timestamp
+	    if (daa_elapsed_time>172800):
+                daa_elapsed_time=172800
+            if (daa_elapsed_time<43200):
+                daa_elapsed_time=43200
+            daa_Wn= (daa_cumulative_work*600)//daa_elapsed_time
+	    daa_target= (2 << 255) // daa_Wn -1
+	    daa_retval = target_to_bits(daa_target)
+	    daa_retval = int(daa_retval)
+            #####FOR TESTING
```

Comment:
Not looked closely at this but indentation looks bad.  Has it been run?  You might want to break out a range chainwork function given a height range, and copy compute_cw_target from mining.py as it's clearer.

Reply:
Yes it runs for me.  The only thing is that it eventually sees bits are different on main net. but we can certainly fix that ident